File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func palindrome(s string) string {
	sArr := strings.Split(s, "")
	for i := 0; i < int(len(sArr)/2); i++ {
		if sArr[i] != sArr[len(sArr)-i-1] {
			return "no"
		}
	}
	return "yes"
}
func main() {
	reader := bufio.NewReader(os.Stdin)
	var s string
	for {
		fmt.Fscanf(reader, "%s\n", &s)
		if s == "0" {
			break
		}
		fmt.Println(palindrome(s))
	}
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

def palindrome(s):
    s_arr = list(s)
    for i in range(len(s_arr)//2):
        if s_arr[i] != s_arr[len(s_arr)-i-1]:
            return "no"
    return "yes"
        
while True:
    s = input().strip('\n')
    if s=='0':
        break
    print(palindrome(s))